pkgs/development/r-modules/default.nix

```diff
@@ -1000,6 +1000,10 @@ let
       preConfigure = "patchShebangs configure";
     });
 
+    gmailr = old.gmailr.overrideAttrs (attrs: {
+      preConfigure = "patchShebangs configure";
```

Comment:
Why not

```suggestion
      postPatch = "patchShebangs configure";
```

?

Reply:
I don't really know, my understanding was that because the configure step was not working, that we'd want to patch the configure script before the configure phase. But your suggestion also works, so I must admit I'm confused, I didn't think there even was a Patch phase.

Reply:
Yes, `postPatch` runs as part of `patchPhase`, therefore running before `configurePhase`.

Reply:
Consistency with the Nixpkgs ecosystem.

Nixpkgs stdenv is structured in a sequence of steps called phases:

https://nixos.org/manual/nixpkgs/stable/#chap-stdenv

As the manual preconizes, the default `phases` should not be modified, because _it is easy to miss some important functionality hidden in some of the less obviously needed phases_.

https://nixos.org/manual/nixpkgs/stable/#ssec-controlling-phases

Therefore we can work with the assumption that each phase is reserved for a certain task.

`patchPhase` deals with changes on the source code - from applying `.patch` files to sed&awk-powered modifications.

`configurePhase` deals with configuring the source tree.

That being said, `patchShebangs` is more appropriate to the patch phase.

Reply:
Thanks for explaining!